api/net/Areas/Admin/Controllers/ChartTemplateController.cs

```diff
@@ -150,8 +150,17 @@ public async Task<IActionResult> PreviewJsonAsync(ChartPreviewRequestModel model
             Template = model.Template,
             SectionSettings = model.Settings
         };
-        var chartTemplate = new TemplateEngine.Models.Reports.ChartEngineContentModel("test", chart, model.Content?.Select(c => new TemplateEngine.Models.ContentModel(c)));
-        var preview = await _reportHelper.GenerateJsonAsync(new TemplateEngine.Models.Charts.ChartRequestModel(chartTemplate, model.ChartData), model.Index, model.Filter, true);
+
+        // Merge content passed to request and also get content from Elasticsearch if a filter is provided.
+        var content = model.Content?.Select(c => new TemplateEngine.Models.ContentModel(c)) ?? Array.Empty<TemplateEngine.Models.ContentModel>();
+        if (model.Filter != null && model.Filter.ToJson() != "{}")
+            content = content.AppendRange(await _reportHelper.FindContentAsync(model.Filter, model.Index));
+
+        // If this chart pulls data from a linked report add this content.
+        var sections = model.LinkedReportId.HasValue ? _reportHelper.GetLinkedReportContent(model.LinkedReportId.Value, null).Result : new();
```

Comment:
Any section that references a linked report needs to fetch that report's content for the current instance.